subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/annotations/InputPropertyAnnotationHandler.java

```diff
@@ -27,11 +28,20 @@
 import java.io.File;
 import java.lang.annotation.Annotation;
 
+import static org.gradle.api.internal.tasks.properties.WorkPropertyAnnotationCategory.OPTIONAL;
+
 public class InputPropertyAnnotationHandler implements PropertyAnnotationHandler {
     public Class<? extends Annotation> getAnnotationType() {
         return Input.class;
     }
 
+    @Override
+    public ImmutableSet<Class<? extends Annotation>> getAllowedModifiers() {
```

Comment:
I expected the allowed categories instead of the allowed supporting annotations. The supported annotations for a type is part of the inspection scheme already, so we only need to limit the categories here, right?

Reply:
That makes sense.